mobile/lib/models/folder/recursive_folder.model.dart

```diff
@@ -0,0 +1,27 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/entities/asset.entity.dart';
+import 'package:immich_mobile/services/folder.service.dart';
+
+class RecursiveFolder {
+  final String name;
+  final String path;
+  List<Asset>? assets;
+  final List<RecursiveFolder> subfolders;
+  final FolderService _folderService;
```

Comment:
I find it strange to include the `FolderService` here. I think the Model should serve solely as a way to represent the data structure and not include dependencies in its structure